src/explorer/SiteTreeItem.ts

```diff
@@ -104,6 +105,9 @@ export abstract class SiteTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
             case WebJobsTreeItem.contextValue:
                 return this._webJobsNode;
             default:
+                if (typeof expectedContextValue === 'string' && DeploymentTreeItem.contextValue.test(expectedContextValue)) {
```

Comment:
Can you explain what you're doing here? I'm a little confused by this

Reply:
For the TreeItemPicker, `SiteTreeItem` wouldn't recognize that `/deployments\//` should return `this.deploymentsNode` since it is only looking for deep equals to the expectedContextValue.

The static `DeploymentTreeItem.contextValue` is a RegExp, but the treeItems in the tree have string context values (ie `deployments/github`).  So in order to recognize that `contextValue` as a deploymentNode, it checks if the `expectedContextValue` matches with the `DeploymentTreeItem` regexp.

Reply:
So `expectedContext` is whatever I pass in to `showTreeItemPicker`.  In this case, it's 'deployment/github' because **only** want deployments that have a github repo associated to it.  

![image](https://user-images.githubusercontent.com/5290572/54717861-63f6e780-4b16-11e9-959c-a3945b1fe56d.png)

Therefore, without that match statement, it looks like this:

![image](https://user-images.githubusercontent.com/5290572/54717887-72450380-4b16-11e9-9f2d-4b46ee47829d.png)

Reply:
Okay makes sense now